heft-plugins/heft-jest-plugin/src/TerminalWritableStream.ts

```diff
@@ -0,0 +1,28 @@
+import { ITerminal } from '@rushstack/node-core-library';
+import { Writable } from 'stream';
+
+// Regex to filter out screen clearing directives
+// eslint-disable-next-line no-control-regex
+const FILTER_REGEX: RegExp = /\x1B\[2J\x1B\[0f|\x1B\[2J\x1B\[3J\x1B\[H/g;
+
+export class TerminalWritableStream extends Writable {
+  private readonly _terminal: ITerminal;
+
+  public constructor(terminal: ITerminal) {
+    super({
+      objectMode: false,
+      decodeStrings: false,
+      defaultEncoding: 'utf-8'
+    });
+
+    this._terminal = terminal;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public _write(chunk: any, encoding: string, callback: (error?: Error | undefined) => void): void {
+    const stringified: string = typeof chunk === 'string' ? chunk : chunk.toString(encoding);
+    const filtered: string = stringified.replace(FILTER_REGEX, '');
+    this._terminal.write(filtered);
```

Comment:
This definitely hits that problem. Likely it will have to go back to not intercepting the calls, or just writing the filtered output (after removing the clear console commands) directly to process.stdout.